man/systemd-repart.xml

```diff
@@ -347,6 +347,106 @@
         if <option>--split</option> is enabled.</para></listitem>
       </varlistentry>
 
+      <varlistentry>
+        <term><option>--include-partition-designators=</option><arg rep="repeat">DESIGNATOR</arg></term>
+        <term><option>--exclude-partition-designators=</option><arg rep="repeat">DESIGNATOR</arg></term>
```

Comment:
let's keep the "designator" name out of the command line arguments. I doubt it's worth throwing this terminology on people. i.e. I think just saying `--include-partition=` is enough.

Reply:
btw, why go by designator anyway? why not go by partition type?

Reply:
One is that it's easier to implement since I can use the bitflags approach with designators. Second is that I if I ever want to skip the root partition or verity or so, I want to be able to do so regardless of the particular architecture that was picked for the root partition, I want a way to skip all root partitions regardless of architecture. Going via designator seems to be the most straightforward way to achieve that.

Reply:
Sure, but partition type and architecture are weirdly intertwined in the same concept. I want to be able to specify a partition's purpose to this option without caring about the architecture. Going by literal partition type doesn't give you that.

Reply:
oh, it does give you that, because the logic to translate partition type names to uuids is arch-aware. so if you just specify "root" and invoke things on x86-64 you'll get the root for x86-64. an on other archs something different...

Reply:
That doesn't work for my use case though. Let's say I'm building an image for aarch64. So the user specifies root-aarch64. On my x86 machine, if I specify exclude "root", the root partition won't actually be excluded.

Frankly I think this logic of automatically converting "root" to the native partition isn't the way to go at all.  I'd split architecture and partition purpose into two separate concepts and synthesize the uuid from that. I'd even add an Architecture field to partition definition files instead of communicating the arch via the type, but that's a discussion for another PR

Reply:
maybe add globs instead? i.e. `--exclude-partition=root-*` would exclude all root* partition types?

Reply:
Sure that would work, but I feel it works around the problem. Why not add designators to the GPT spec instead? Maybe rename it into "purpose" instead of "designator". Each GPT Type UUID could then have a purpose and an optional architecture, and in various tools, you could filter by purpose or by architecture.

Reply:
mostly because designators just complicate stuff. i kinda want to keep the concept internal to systemd